external/boost/archive/portable_binary_iarchive.hpp

```diff
@@ -258,7 +258,7 @@ portable_binary_iarchive::load_impl(boost::intmax_t & l, char maxsize){
     this->primitive_base_t::load_binary(cptr, size);
 
     #ifdef BOOST_BIG_ENDIAN
-        if(m_flags & endian_little)
+        if((m_flags & endian_little) || (!(m_flags & endian_big)))

```

Comment:
I cannot comment on this below - when loading flags, if neither `endian_little` nor `endian_big` is set, should it set `endian_little` ? This would enable a big endian machine to load existing portable archives. It seems unlikely that any big endian archives were ever written.

Reply:
I guess it's fair since it's not used as generic code. I'll do that.

Reply:
Something like this ?

```
diff --git a/external/boost/archive/portable_binary_iarchive.hpp b/external/boost/archive/portable_binary_iarchive.hpp
index 88a9279e7..ec3ac8f67 100644
--- a/external/boost/archive/portable_binary_iarchive.hpp
+++ b/external/boost/archive/portable_binary_iarchive.hpp
@@ -343,6 +343,8 @@ portable_binary_iarchive::init(unsigned int flags){
         );
         #endif
     }
+    if (!(m_flags & (endian_little | endian_big)))^M
+      m_flags |= endian_little;^M
     unsigned char x;
     load(x);
     m_flags = x << CHAR_BIT;

```

Reply:
Yeah that would work. Flags can still be specified, but you cannot omit an endianness flag.